From 35fb81cbb44baea24fc27c1d1c0ce992bd36a91a Mon Sep 17 00:00:00 2001
From: Han Xu <han.xu@nxp.com>
Date: Thu, 11 Oct 2018 15:32:04 -0500
Subject: [PATCH 4821/5242] MLK-19897: dma: mxs-dma: update mxs-dma filter
 function to filter out the unrelated dma channels

commit  faf31389b85598daed01920fcfb344c4cf4ac3a1 from
https://source.codeaurora.org/external/imx/linux-imx.git

update mxs-dma filter function to firstly filter the dma channels only for mxs-dma, rather than checking unrelated dma chans in following code.

Signed-off-by: Han Xu <han.xu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/mxs-dma.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/dma/mxs-dma.c b/drivers/dma/mxs-dma.c
index 7507d29..3602f4e 100644
--- a/drivers/dma/mxs-dma.c
+++ b/drivers/dma/mxs-dma.c
@@ -775,6 +775,9 @@ static bool mxs_dma_filter_fn(struct dma_chan *chan, void *fn_param)
 	struct mxs_dma_engine *mxs_dma = mxs_chan->mxs_dma;
 	int chan_irq;
 
+	if (strcmp(chan->device->dev->driver->name, "mxs-dma"))
+		return false;
+
 	if (!mxs_dma)
 		return false;
 
-- 
1.7.9.5

